drivers/sensor/ti/lm95234/lm95234.c

```diff
@@ -0,0 +1,321 @@
+#include <zephyr/drivers/sensor/lm95234.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(LM95234, CONFIG_SENSOR_LOG_LEVEL);
+
+#define LM95234_REG_LOCAL_TEMP_SIGNED_MSB				0x10
+#define LM95234_REG_LOCAL_TEMP_SIGNED_LSB				0x20
+#define LM95234_REG_REMOTE_TEMP_1_SIGNED_MSB				0x11
+#define LM95234_REG_REMOTE_TEMP_1_SIGNED_LSB				0x21
+#define LM95234_REG_REMOTE_TEMP_2_SIGNED_MSB				0x12
+#define LM95234_REG_REMOTE_TEMP_2_SIGNED_LSB				0x22
+#define LM95234_REG_REMOTE_TEMP_3_SIGNED_MSB				0x13
+#define LM95234_REG_REMOTE_TEMP_3_SIGNED_LSB				0x23
+#define LM95234_REG_REMOTE_TEMP_4_SIGNED_MSB				0x14
+#define LM95234_REG_REMOTE_TEMP_4_SIGNED_LSB				0x24
+#define LM95234_REG_REMOTE_TEMP_1_UNSIGNED_MSB				0x19
+#define LM95234_REG_REMOTE_TEMP_1_UNSIGNED_LSB				0x29
+#define LM95234_REG_REMOTE_TEMP_2_UNSIGNED_MSB				0x1a
+#define LM95234_REG_REMOTE_TEMP_2_UNSIGNED_LSB				0x2a
+#define LM95234_REG_REMOTE_TEMP_3_UNSIGNED_MSB				0x1b
+#define LM95234_REG_REMOTE_TEMP_3_UNSIGNED_LSB				0x2b
+#define LM95234_REG_REMOTE_TEMP_4_UNSIGNED_MSB				0x1c
+#define LM95234_REG_REMOTE_TEMP_4_UNSIGNED_LSB				0x2c
+#define LM95234_REG_DIODE_MODEL_SELECT					0x30
+#define LM95234_REG_REMOTE_1_OFFSET					0x31
+#define LM95234_REG_REMOTE_2_OFFSET					0x32
+#define LM95234_REG_REMOTE_3_OFFSET					0x33
+#define LM95234_REG_REMOTE_4_OFFSET					0x34
+#define LM95234_REG_CONFIG						0x03
+#define LM95234_REG_CONV_RATE						0x04
+#define LM95234_REG_CHANNEL_CONV_ENABLE					0x05
+#define LM95234_REG_FILTER_SETTING					0x06
+#define LM95234_REG_ONESHOT						0x0f
+#define LM95234_REG_COMMON_STATUS					0x02
+#define LM95234_REG_STATUS_1						0x07
+#define LM95234_REG_STATUS_2						0x08
+#define LM95234_REG_STATUS_3						0x09
+#define LM95234_REG_STATUS_4						0x0a
+#define LM95234_REG_DIODE_MODEL_STATUS					0x38
+#define LM95234_REG_TCRIT1_MASK						0x0c
+#define LM95234_REG_TCRIT2_MASK						0x0d
+#define LM95234_REG_TCRIT3_MASK						0x0e
+#define LM95234_REG_LOCAL_TCRIT_LIMIT					0x40
+#define LM95234_REG_REMOTE1_TCRIT1_LIMIT				0x41
+#define LM95234_REG_REMOTE2_TCRIT1_LIMIT				0x42
+#define LM95234_REG_REMOTE3_TCRIT_LIMIT					0x43
+#define LM95234_REG_REMOTE4_TCRIT_LIMIT					0x44
+#define LM95234_REG_REMOTE1_TCRIT23_LIMIT				0x49
+#define LM95234_REG_REMOTE2_TCRIT23_LIMIT				0x4a
+#define LM95234_REG_COMMON_TCRIT_HYSTERESIS				0x5a
+#define LM95234_REG_MANUF_ID						0xfe
+#define LM95234_REG_REV_ID						0xff
+
+#define LM95234_MAN_ID							0x01
+#define LM95234_CHIP_ID							0x79
+
+#define LM95234_CONFIG_STANDBY						BIT(6)
+
+struct lm95234_data {
+	/** Temperatures in raw format read from sensor */
+	int32_t local;
+	int32_t remote[4];
+};
+
+struct lm95234_config {
+	struct i2c_dt_spec i2c;
+};
+
+static inline int lm95234_fetch_temp(const struct lm95234_config *cfg, struct lm95234_data *data,
+				     enum sensor_channel chan, int32_t *output)
+{
+	int ret;
+	uint8_t val;
+
+	*output = 0;
```

Comment:
I was not clear in my initial comment.
What I think can potentially be a problem is that `lm95234_data` value is set to 0 if something fails and a previous valid value is overwritten.
What I think could be improved is that `lm95234_channel_get` could either give the previous valid value or an error if there a new value could not be fetched.
I might be overthinking this, since you probably only call `get` if `fetch` succeeds.

Reply:
Ahh, I see what you mean. It's probably not a huge issue, but not hard to improve.

Reply:
Changed to only overwrite the output value on success.